overlord/snapstate/snapstate.go

```diff
@@ -2688,7 +2696,23 @@ func RemoveMany(st *state.State, names []string) ([]string, []*state.TaskSet, er
 	return removed, tasksets, nil
 }
 
-func validateSnapNames(names []string) error {
+// validateSnapNames returns an error if a snap name
+// (only one condition needs to be met):
+//	* differs from the expected format
+//	* is duplicated in the list
+func validateSnapNames(op string, names []string) error {
+	if invalidNames := findInvalidSnapNames(names); len(invalidNames) > 0 {
+		return fmt.Errorf("cannot %s due to invalid snap names: %v", op, strings.Join(invalidNames, ", "))
```

Comment:
Why aren't you just passing `invalidNames` to `fmt.Errorf`? It should be doing the string concatenation itself.

Reply:
Thought it looked simpler without the brackets around the list. I can change it though, it's not a strong opinion